flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/delegation/PlannerBase.scala

```diff
@@ -176,11 +176,11 @@ abstract class PlannerBase(
         val input = getRelBuilder.queryOperation(modifyOperation.getChild).build()
         val identifier = catalogManager.qualifyIdentifier(catalogSink.getTablePath: _*)
         getTableSink(identifier).map(sink => {
-          TableSinkUtils.validateSink(catalogSink, identifier, sink)
+          val partKeys =
+            catalogManager.getTable(identifier).get().asInstanceOf[CatalogTable].getPartitionKeys
```

Comment:
Should we check the catalog type before cast to `CatalogTable`? For example, it would be a `CatalogView` when get table from catalog manager.

Reply:
`getTableSink(identifier)`  already has this check, I will change the return type of `getTableSink` and get `CatalogTable` directly.